### PR TITLE
Feat/add action mailer

### PR DESCRIPTION
**Pull request by piowit**

### CODE REVIEW

**Comment by piowit** [comment]
You can test sending emails by creating an account with your email in app, then loging in as admin and delivering an order (you have previously add kudo and buy reward)
https://employeerecognitionplatfrom.herokuapp.com/

File: app/controllers/admin/orders_controller.rb [hunk 1]
```diff
@@ -5,19 +5,5 @@ class OrdersController < AdminController
...
-        redirect_to admin_orders_path, notice: 'Order already delivered'
-      else
-        @order.delivered = true
-        if @order.save
-          redirect_to admin_orders_path, notice: 'Order delivered'
-        else
-          redirect_to admin_orders_path, notice: 'Error'
-        end
-      end
-    end
```
Comment (piowit) [inline app/controllers/admin/orders_controller.rb:21]: forgot to delete this earlier

File: config/routes.rb [hunk 2]
```diff
@@ -16,10 +16,6 @@
     resources :company_values
     resources :rewards
     resources :deliveries, only: %i[update]
-    resources :orders, only: %i[index] do
-      member do
-        get 'deliver'
-      end
-    end
```
Comment (piowit) [inline config/routes.rb:23]: forgot to delete this earlier

File: app/controllers/admin/deliveries_controller.rb [hunk 3]
```diff
@@ -9,6 +9,7 @@ def update
       else
         @order.delivered = true
         if @order.save
+          DeliveryOrderMailer.send_delivery_confirmation_email(@order).deliver_now
```
Comment (oskarlakner) [inline app/controllers/admin/deliveries_controller.rb]: I'd recommend using the new syntax with: `.with(`

File: config/environments/production.rb [hunk 4]
```diff
@@ -55,7 +55,19 @@
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
+  ActionMailer::Base.smtp_settings = {
+    user_name: 'apikey', # This is the string literal 'apikey', NOT the ID of your API key
+    password: Rails.application.credentials.sendgrid[:secret_access_key], # This is the secret sendgrid API key which was issued during API key creation
+    domain: 'employeerecognitionplatfrom.heroku.com',
```
Comment (oskarlakner) [inline config/environments/production.rb:61]: this should be in credentials too

File: spec/mailers/previews/delivery_order_mailer_preview.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class DeliveryOrderMailerPreview < ActionMailer::Preview
+  def send_delivery_confirmation_email
+    DeliveryOrderMailer.send_delivery_confirmation_email(FactoryBot.build(:order))
```
Comment (oskarlakner) [inline spec/mailers/previews/delivery_order_mailer_preview.rb]: 👏

**Review by oskarlakner — APPROVED** [review]

**Pull request merged**